MapboxDirections/MBSpeedLimit.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  Created by Avi Cieplinski on 4/15/19.
+//  Copyright © 2019 Mapbox. All rights reserved.
+//
+
+import Foundation
+
+@objc(MBSpeedLimit)
+public class SpeedLimit: NSObject {
+
+    @objc public static let invalid = Double(-1)
+
+    /**
+     A unitless measure of speed which is dependent on the `SpeedLimit.speedUnit`.
+
+     By default, the speed will be unknown and equal to `MBSpeedIsInvalid` which is -1.
+
+     If the speed is none, the value will be equal to `Double.greatestFiniteMagnitude`.
+     */
+    @objc public var value: Double = invalid
+
+
+    /**
+     Units for `SpeedLimit.speed`.
+     */
+    @objc public var unit: SpeedUnit = .kilometersPerHour
+
+    /**
+     Initialize a new `SpeedLimit` object.
+     */
+    public init(value: Double, speedUnit: SpeedUnit) {
+        self.value = value
+        self.unit = speedUnit
+    }
+
+    /**
+     Initialize a new `SpeedLimit` object from a JSON dictionary.
+     */
+    @objc public convenience init(json: [String: Any]) {
+        var speed = json["speed"] as? Double ?? SpeedLimit.invalid
+        if let speedNone = json["unknown"] as? Bool, speedNone {
+            speed = .greatestFiniteMagnitude
+        }
+
+        let speedUnit: SpeedUnit
+        if let speedUnitString = json["unit"] as? String {
+            speedUnit = SpeedUnit(description: speedUnitString) ?? .kilometersPerHour
+        } else {
+            speedUnit = .kilometersPerHour
+        }
+
+        self.init(value: speed, speedUnit: speedUnit)
+    }
+
+    public static var supportsSecureCoding = true
+
+    public required init?(coder decoder: NSCoder) {
+        value = decoder.decodeDouble(forKey: "value")
+
+        guard let speedUnitString = decoder.decodeObject(of: NSString.self, forKey: "unit") as String?, let speedUnit = SpeedUnit(description: speedUnitString) else {
+            return nil
+        }
+        self.unit = speedUnit
+    }
+
+    public func encode(with coder: NSCoder) {
+        coder.encode(value, forKey: "value")
+        coder.encode(unit, forKey: "unit")
+    }
+}
+
+/**
+ Unit of measurement for `SpeedLimit`.
+ */
+@objc(MBSpeedUnit)
+public enum SpeedUnit: Int, CustomStringConvertible {
+
+    /**
+     Indicates the road segment uses miles per hour for measuring speed limits.
+     */
+    case milesPerHour
+
+    /**
+     Indicates the road segment uses kilometers per hour for measuring speed limits.
+     */
+    case kilometersPerHour
+
+    public init?(description: String) {
+        let unit: SpeedUnit
+        switch description {
+        case "mph":
+            unit = .milesPerHour
+        case "km/h":
+            unit = .kilometersPerHour
+        default:
+            unit = .kilometersPerHour
+        }
+        self.init(rawValue: unit.rawValue)
+    }
+
+    public var description: String {
+        switch self {
+        case .milesPerHour:
+            return "mph"
+        case .kilometersPerHour:
+            return "kph"
```

Comment:
The Directions API seems to use `km/h` rather than `kph` to indicate kilometers per hour. The initializer is already using `km/h`, so this property doesn’t round-trip.

```suggestion
            return "km/h"
```